src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Boolean.cs

```diff
@@ -312,6 +321,92 @@ internal override IR.Nodes.IntermediateNode CreateIRCallNode(PowerFx.Syntax.Call
                 return base.CreateIRCallNode(node, context, args, scope);
             }
         }
+    }
+
+    // Boolean(arg:L)
+    // Corresponding Excel and DAX function: Boolean
+    internal sealed class BooleanLFunction : BuiltinFunction
+    {
+        public override bool IsSelfContained => true;
+
+        public override bool SupportsParamCoercion => false;
+
+        public BooleanLFunction()
+            : base(BooleanFunction.BooleanInvariantFunctionName, TexlStrings.AboutBooleanL, FunctionCategories.Text, DType.Boolean, 0, 1, 1, DType.OptionSetValue)
+        {
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.BooleanLArg1 };
+        }
+
+        public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            var fValid = base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+
+            var arg = args[0];
```

Comment:
nit, may be not needed here ?

Reply:
Good point.  I cleaned up the code, tried to make clearer.